src/java.base/share/classes/java/lang/template/StringTemplate.java

```diff
@@ -0,0 +1,351 @@
+            throw new IllegalArgumentException(
+                    "fragments list size is not one more than values list size");
+        }
+        return TemplateSupport.of(fragments, values);
+    }
+
+    /**
+     * Creates a string that interleaves the elements of values between the
+     * elements of fragments.
+     *
+     * @param fragments  list of String fragments
+     * @param values     list of expression values
+     *
+     * @return String interpolation of fragments and values
+     *
+     * @throws IllegalArgumentException if fragments list size is not one more
+     *         than values list size
+     * @throws NullPointerException fragments or values is null or if any of the fragments is null
+     */
+    static String interpolate(List<String> fragments, List<?> values) {
+        Objects.requireNonNull(fragments, "fragments must not be null");
+        Objects.requireNonNull(values, "values must not be null");
+        int fragmentsSize = fragments.size();
+        int valuesSize = values.size();
+        if (fragmentsSize != valuesSize + 1) {
+            throw new IllegalArgumentException("fragments must have one more element than values");
+        }
+        return TemplateSupport.interpolate(fragments, values);
+    }
+
+    /**
+     * Combine zero or more {@link StringTemplate StringTemplates} into a single
+     * {@link StringTemplate}.
+     * {@snippet :
+     * StringTemplate st = StringTemplate.combine(RAW."\{a}", RAW."\{b}", RAW."\{c}");
+     * assert st.interpolate().equals(RAW."\{a}\{b}\{c}");
+     * }
+     * Fragment lists from each {@link StringTemplate} are merged such that the last fragment
+     * from the previous {@link StringTemplate} is concatenated with the first fragment of the next
+     * {@link StringTemplate}. Values lists are simply concatenated to produce a single values list.
+     * The result is a well-formed {@link StringTemplate} with n+1 fragments and n values, where
+     * n is the total of number of values across all the supplied
+     * {@link StringTemplate StringTemplates}.
+     *
+     * @param stringTemplates  zero or more {@link StringTemplate}
+     *
+     * @return combined {@link StringTemplate}
+     *
+     * @throws NullPointerException if stringTemplates is null or if any of the elements are null
+     *
+     * @implNote If zero {@link StringTemplate} arguments are provided then a
+     * {@link StringTemplate} with an empty fragment and no values is returned, as if invoking
+     * <code>StringTemplate.of("")</code> . If only one {@link StringTemplate} argument is provided
+     * then it is returned unchanged.
+     */
+    static StringTemplate combine(StringTemplate... stringTemplates) {
+        return TemplateSupport.combine(stringTemplates);
+    }
+
+    /**
+     * This {@link StringProcessor} instance is conventionally used for the string interpolation
+     * of a supplied {@link StringTemplate}.
+     * <p>
```

Comment:
It should be mentioned that the string representations are created as if invoking {@link String#valueOf}.
An perhaps a link/\@see to `FMT` for control over formatting.  

Also include it in the javadoc for `interpolate`.

Reply:
Changing.